File: packages/cli/test/fixtures/multilang/sample.cpp
#include <string>
#include <map>
#include <iostream>

class User {
public:
    std::string id;
    std::string name;

    User(const std::string& id, const std::string& name)
        : id(id), name(name) {}

    std::string full_name() const {
        return "User: " + name;
    }
};

class UserStore {
private:
    std::map<std::string, User> users;

public:
    void save(const User& user) {
        users[user.id] = user;
    }

    User* find(const std::string& id) {
        auto it = users.find(id);
        if (it != users.end()) return &it->second;
        return nullptr;
    }
};

std::string generate_id() {
    return "uuid-placeholder";
}

User create_user(const std::string& name) {
    return User(generate_id(), name);
}

File: packages/cli/test/fixtures/multilang/sample.rs
use std::collections::HashMap;

pub struct UserStore {
    users: HashMap<String, User>,
}

pub struct User {
    pub id: String,
    pub name: String,
}

pub trait Repository {
    fn find(&self, id: &str) -> Option<&User>;
    fn save(&mut self, user: User);
}

impl Repository for UserStore {
    fn find(&self, id: &str) -> Option<&User> {
        self.users.get(id)
    }

    fn save(&mut self, user: User) {
        self.users.insert(user.id.clone(), user);
    }
}

pub fn create_user(name: &str) -> User {
    User {
        id: generate_id(),
        name: name.to_string(),
    }
}

fn generate_id() -> String {
    String::from("uuid-placeholder")
}

pub enum UserRole {
    Admin,
    Member,
    Guest,
}

pub type UserId = String;
